Makefile

```diff
@@ -139,13 +142,61 @@ help-install:
 #
 ###############################################################
 
+### Deployment ###############################################
+#
+release-check:
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	 if test -z "$$VERSION"; then                     \
+		echo VERSION is not set.;                 \
+		echo Please create a git tag;             \
+		exit 100;                                 \
+	else                                              \
+		echo "Releasing version: $$VERSION";      \
+	fi;
+
+release-prep: release-check
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	python setup.py clean;                            \
+	python setup.py sdist;                            \
+	pip install dist/starfish-$$VERSION.tar.gz
+
+release-verify: release-check slow
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	docker tag $(DOCKER_IMAGE) $(DOCKER_IMAGE):$$VERSION && \
+	docker tag $(DOCKER_IMAGE) $(DOCKER_IMAGE):$$VERSION-$(DOCKER_BUILD)
+
+release-upload: release-check
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	printf '\n# Please execute the following steps\n';\
+	echo git push origin $$VERSION;                   \
+	echo docker push $(DOCKER_IMAGE);                 \
```

Comment:
Who owns the docker image rights?  I presume not everyone can just push it?

Reply:
At the moment, I do. Unfortunately, it can't be synced with github teams. I'll pass over to whomever. Can you confirm that you are also `@ttung` on Docker Hub?

Reply:
I don't know.  :)

Once this lands, I think it would make sense to distribute it to everyone who regularly commits here (yourself, me, @ambrosejcarr, @shanaxel42, @kne42)

Reply:
I am ambrosejcarr on dockerhub.